src/socket/tcp_server.cr

```diff
@@ -22,15 +22,12 @@ class TCPServer < TCPSocket
     super(family)
   end
 
-  def initialize(host : String, port : Int, backlog = SOMAXCONN, dns_timeout = nil)
+  def initialize(host : String, port : Int, backlog = SOMAXCONN, dns_timeout = nil, reuse_port = false)
     Addrinfo.tcp(host, port, timeout: dns_timeout) do |addrinfo|
       super(addrinfo.family, addrinfo.type, addrinfo.protocol)
 
       self.reuse_address = true
-      begin
-        self.reuse_port = true
-      rescue Errno
-      end
+      self.reuse_port = true if reuse_port
```

Comment:
Bonus: now we can raise if the OS doesn't support `SO_REUSEPORT`, since it's now explicitly enabled.